scripts/idl/generators/types.py

```diff
@@ -195,7 +195,7 @@ def is_struct(self) -> bool:
     "command_id": BasicInteger(idl_name="command_id", byte_count=4, is_signed=True),
     "data_ver": BasicInteger(idl_name="data_ver", byte_count=4, is_signed=True),
     "date": BasicInteger(idl_name="date", byte_count=4, is_signed=True),
-    "devtype_id": BasicInteger(idl_name="devtype_id", byte_count=4, is_signed=True),
+    "devtype_id": BasicInteger(idl_name="devtype_id", byte_count=4, is_signed=False),
     "endpoint_no": BasicInteger(idl_name="endpoint_no", byte_count=2, is_signed=True),
```

Comment:
Note that there are more unsigned types here, e.g. CommandId, EndpointNo etc.

Reply:
Will update in #23017  .. this was intended as hotfix to fix a compile error, but for now we may as well fix the rest.